### PR TITLE
Remove Phoenix4 connector

### PR DESCRIPTION
**Pull request by wendigo**
## Description

As discussed with https://github.com/orgs/trinodb/teams/maintainers over slack thread https://trinodb.slack.com/archives/CP1MUNEUX/p1654514925872899

> Is this change a fix, improvement, new feature, refactoring, or other?

Decline :)

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Connector

> How would you describe this change to a non-technical end user or system administrator?

Remove support for connecting to Phoenix 4 cluster

## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
(x) Release notes entries required with the following suggested text:

```markdown
# General
* Remove support for Phoenix 4
```

### CODE REVIEW

**Comment by losipiuk** [comment]
Please add release notes using PR message template

**Comment by losipiuk** [comment]
Are there some docs changes needed?

**Comment by findepi** [comment]
LGTM, but there are conflicts.

**Review by electrum — APPROVED** [review]

**Comment by wendigo** [comment]
@findepi rebased. Sorry for that (old master)

**Comment by wendigo** [comment]
And docs updated.

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodePhoenix4.java [hunk 1]
```diff
@@ -1,58 +0,0 @@
...
-public final class EnvMultinodePhoenix4
-        extends EnvironmentProvider
-{
-    private final ResourceProvider configDir;
-
-    @Inject
-    public EnvMultinodePhoenix4(StandardMultinode standardMultinode, Phoenix phoenix, DockerFiles dockerFiles)
-    {
-        super(standardMultinode, phoenix);
-        this.configDir = requireNonNull(dockerFiles, "dockerFiles is null").getDockerFilesHostDirectory("conf/environment/multinode-phoenix4/");
```
Comment (ebyhr) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodePhoenix4.java:42]: Could you remove `multinode-phoenix4` directory?

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/Suite6NonGeneric.java [hunk 2]
```diff
@@ -59,9 +58,6 @@ public List<SuiteTestRun> getTestRuns(EnvironmentConfig config)
                 testOnEnvironment(EnvMultinodeKafkaSaslPlaintext.class)
                         .withGroups("configured_features", "kafka")
                         .build(),
-                testOnEnvironment(EnvMultinodePhoenix4.class)
```
Comment (ebyhr) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/Suite6NonGeneric.java:62]: Could you remove `phoenix` from `EnvMultinodeAllConnectors`?

**Review by ebyhr — APPROVED** [review]

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodePhoenix4.java (hunk 1 above)
Comment (wendigo) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodePhoenix4.java:42]: Done

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/Suite6NonGeneric.java (hunk 2 above)
Comment (wendigo) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/suite/suites/Suite6NonGeneric.java:62]: Done

**Pull request merged**